packages/proxy/lib/http/util/cookies.ts

```diff
@@ -0,0 +1,166 @@
+export const parseCookie = (cookie) => {
+  const toughCookie = CookieJar.parse(cookie)
+
+  if (!toughCookie) return
+
+  // fixes tough-cookie defaulting undefined/invalid SameSite to 'none'
+  // https://github.com/salesforce/tough-cookie/issues/191
+  const hasUnspecifiedSameSite = toughCookie.sameSite === 'none' && !sameSiteNoneRe.test(cookie)
+
+  // not all browsers currently default to lax, but they're heading in that
+  // direction since it's now the standard, so this is more future-proof
+  if (hasUnspecifiedSameSite) {
+    toughCookie.sameSite = 'lax'
+  }
+
+  return toughCookie
+}
+
+const comparableCookieString = (toughCookie) => {
+  return _(toughCookie)
+  .pick('key', 'value', 'domain', 'path')
+  .toPairs()
+  .sortBy(([key]) => key)
+  .map(([key, value]) => `${key}=${value}`)
+  .join('; ')
+}
+
+const areCookiesEqual = (cookieA, cookieB) => {
+  return comparableCookieString(cookieA) === comparableCookieString(cookieB)
+}
+
+const matchesPreviousCookie = (previousCookies, cookie) => {
+  return !!previousCookies.find((previousCookie) => {
+    return areCookiesEqual(previousCookie, cookie)
+  })
+}
+
+const toughCookieToAutomationCookie = (toughCookie, defaultDomain) => {
+  const expiry = toughCookie.expiryTime()
+
+  return {
+    domain: toughCookie.domain || defaultDomain,
+    expiry: isFinite(expiry) ? expiry / 1000 : null,
+    httpOnly: toughCookie.httpOnly,
+    maxAge: toughCookie.maxAge,
+    name: toughCookie.key,
+    path: toughCookie.path,
+    sameSite: toughCookie.sameSite === 'none' ? 'no_restriction' : toughCookie.sameSite,
+    secure: toughCookie.secure,
+    value: toughCookie.value,
+  }
+}
+
+/**
+ * Utility for dealing with cross-origin cookies
+ * - Tracks which cookies were added to our server-side cookie jar during
+ *   a request, so they can be added to the browser via automation
+ * - Provides utility cookie-handling methods that rely on aspects of the
+ *   request (url, previous request url, etc)
+ */
+export class CookiesHelper {
+  cookieJar: CookieJar
+  currentAUTUrl: string | undefined
+  request: RequestDetails
+  debug: Debug.Debugger
+  defaultDomain: string
+  sameSiteContext: 'strict' | 'lax' | 'none'
+  previousCookies: Cookie[] = []
+
+  constructor ({ cookieJar, currentAUTUrl, request, debug }) {
+    this.cookieJar = cookieJar
+    this.currentAUTUrl = currentAUTUrl
+    this.request = request
+    this.debug = debug
+    this.sameSiteContext = getSameSiteContext(currentAUTUrl, request.url, request.isAUTFrame)
+
+    const parsedRequestUrl = new URL(request.url)
+
+    this.defaultDomain = parsedRequestUrl.hostname
+  }
+
+  async capturePreviousCookies () {
+    // this plays a part in adding cross-origin cookies to the browser via
+    // automation. if the request doesn't need cross-origin handling, this
+    // is a noop
+    if (!this.request.needsCrossOriginHandling) return
+
+    this.previousCookies = this.cookieJar.getAllCookies()
+  }
+
+  async getAddedCookies () {
```

Comment:
```suggestion
 getAddedCookies () {
```

Reply:
Addressed in [`95124a3` (#22320)](https://github.com/cypress-io/cypress/pull/22320/commits/95124a38297f4924a8a11dec21668171fa50d514)